src/Codeception/Subscriber/ErrorHandler.php

```diff
@@ -80,6 +87,14 @@ public function errorHandler($errno, $errstr, $errfile, $errline, $context = arr
             return false;
         }
 
+        if($errno === E_USER_NOTICE) {
+            throw new Notice($errstr, $errno,  $errfile, $errline);
+        }elseif ($errno === E_USER_WARNING) {
+            throw new Warning($errstr, $errno,  $errfile, $errline);
+        }elseif ($errno === E_USER_ERROR) {
+            throw new Error($errstr, $errno,  $errfile, $errline);
+        }
+        
```

Comment:
To get this change into Codeception 4, it must be compatible with PHPUnit 5.7.
Classes you used here were renamed in PHPUnit 6.0, so this code would cause fatal error.
Please add some test that runs on all versions of PHP.
Then add class aliases to https://github.com/Codeception/phpunit-wrapper/blob/6.0/src/shim.php

Reply:
@Naktibalda What about the deprecation stuff? 

Codeception should support `expectDeprecation` aswell.

But I don't understand what is done there currently. Looks like something symfony-related.

Reply:
Codeception doesn't have to support any of this.

Deprecations are counted and summary is printed below test results:
![image](https://user-images.githubusercontent.com/395992/169759773-a820b54c-35c4-46c7-8d73-d3abe6695882.png)

Reply:
> Codeception doesn't have to support any of this.
> 
> Deprecations are counted and summary is printed below test results: ![image](https://user-images.githubusercontent.com/395992/169759773-a820b54c-35c4-46c7-8d73-d3abe6695882.png)

But doesn't this just show the deprecations that occurred while running the test?

What I'm talking about (and what PHPUnit supports in the `TestCase` class) is making assertions about deprecations.

```php

class Foo {

   public function foo(): {
     trigger_error('foo is deprecated', E_USER_DEPRECATED);
   }

}

class FooTest extends Codeception\Unit {

  public function test_that_foo_triggers_deprecation() {
  
     $this->expectDeprecation('foo is deprecated');
     
     (new Foo())->foo();
     
  }


}

```

Reply:
Since 8.5.21 PHPUnit doesn't convert deprecations to exceptions by default, it requires setting configuration option.

https://github.com/sebastianbergmann/phpunit/blob/8.5/ChangeLog-8.5.md#8521---2021-09-25

You could implement `convert_deprecations_to_exceptions` option too.

* See how `error_level` option is handled in `handle`.
* Add new option to `$defaultSuiteSettings` https://github.com/Codeception/Codeception/blob/4.2/src/Codeception/Configuration.php#L116-L137 (the default value should be false too).
* Document at https://github.com/Codeception/codeception.github.com/blob/master/docs/reference/Configuration.md#settings